app/layer_generators/main.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+
+	"golang.org/x/tools/imports"
+)
+
+var reserved = []string{"AcceptLanguage", "AccountMigration", "Cluster", "Compliance", "Context", "DataRetention", "Elasticsearch", "HTTPService", "ImageProxy", "IpAddress", "Ldap", "Log", "MessageExport", "Metrics", "Notification", "NotificationsLog", "Path", "RequestId", "Saml", "Session", "SetIpAddress", "SetRequestId", "SetSession", "SetStore", "SetT", "Srv", "Store", "T", "Timezones", "UserAgent", "SetUserAgent", "SetAcceptLanguage", "SetPath", "SetContext", "SetServer", "GetT"}
+
+var outputFile string
+var inputFile string
+var outputFileTemplate string
+
+const OPEN_TRACING_PARAMS_MARKER = "@openTracingParams"
+
+func main() {
+	flag.StringVar(&inputFile, "in", path.Join("..", "app_iface.go"), "App interface file")
+	flag.StringVar(&outputFile, "out", path.Join("..", "opentracing_layer.go"), "Output file")
+	flag.StringVar(&outputFileTemplate, "template", "opentracing_layer.go.tmpl", "Output template file")
+	flag.Parse()
+	BuildOpenTracingAppLayer()
+}
+
+func BuildOpenTracingAppLayer() {
+	code := GenerateLayer("OpenTracingAppLayer", outputFileTemplate)
+	formattedCode, err := imports.Process(outputFile, []byte(code), &imports.Options{Comments: true})
+	if err != nil {
+		panic(err)
```

Comment:
Let's avoid panics here and everywhere else. We can bubble up the error with appropriate context, and call a log.Fatal in the main if we need to exit immediately. Else a simple log.Println and return.